applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,383 @@
+6. [SynthesizeAudio](#SynthesizeAudio) - synthesizes human-like audio from text
+7. `DynamoDB:PutItem` (see [table config](#managing-items-in-dynamodb))
+8. Stop
+
+The below diagram depicts this sequence.
+![state-machine.png](state-machine.png)
+
+## State machine Lambda functions
+
+See below for the required inputs and outputs of each Lambda function.
+
+### ExtractText
+Uses Amazon Textract's [DetectDocumentText](https://docs.aws.amazon.com/textract/latest/dg/API_DetectDocumentText.html)
+method to extract text from an image and return a unified text representation.
+
+#### **Input**
+
+Uses the data available on the [S3 event object](https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventBridge.html).
+
+For example:
+
+```json
+{
+  "bucket": "my-s3-bucket",
+  "region": "us-east-1",
+  "object": "foo/bar.png"
+}
+```
+
+#### **Output**
+
+Returns a string representing the text extracted.
+
+For example:
+
+```
+CET HÔTEL ÉTAIT SUPER
+```
+---
+### AnalyzeSentiment
+Uses Amazon Comprehend's [DetectSentiment](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_DetectSentiment.html)
+method to detect sentiment (`POSITIVE`, `NEUTRAL`, `MIXED`, or `NEGATIVE`).
+
+#### **Input**
+
+Uses the data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+
+```json
+{
+  "source_text": "CET HÔTEL ÉTAIT SUPER",
+  "region": "us-east-1"
+}
+```
+
+#### **Output**
+
+Returns the guessed sentiment and language code.
+
+For example:
+
+```json
+{
+  "sentiment": "POSITIVE",
+  "language_code": "fr-FR"
+}
+```
+---
+### TranslateText
+Uses Amazon Translate's [TranslateText](https://docs.aws.amazon.com/translate/latest/APIReference/API_TranslateText.html)
+method to translate text to English and return a string.
+
+#### **Input**
+
+Uses the data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+
+```json
+{
+    "source_language_code": "fr-FR",
+    "region": "us-east-1",
+    "extracted_text": "CET HÔTEL ÉTAIT SUPER"
+}
+```
+
+#### **Output**
+
+Returns a string representing the translated text.
+
+For example:
+
+```
+THIS HOTEL WAS GREAT
+```
+---
+### SynthesizeAudio
+Uses Amazon Polly's [SynthesizeAudio](https://docs.aws.amazon.com/polly/latest/dg/API_SynthesizeSpeech.html)
+method to converts input text into life-like speech
+
+#### **Input**
+
+Uses the data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+
+```json
+{
+  "bucket": "my-s3-bucket",
+  "translated_text": "THIS HOTEL WAS GREAT",
+  "region": "us-east-1",
+  "object": "comment.png"
+}
+```
+
+#### **Output**
+
+Returns a string representing the key of the synthesized audio file.
+
+For example:
+
+```
+my-s3-bucket/audio.mp3
```

Comment:
Here and elsewhere, recommend using DOC-EXAMPLE-BUCKET as the example bucket name.